repose-aggregator/extensions/simple-rbac/src/main/scala/org/openrepose/filters/simplerbac/SimpleRbacFilter.scala

```diff
@@ -0,0 +1,294 @@
+    val dispatchResultHandler = new DispatchResultHandler
+    Option(configuration.getDelegating) match {
+      case Some(delegating) =>
+        dispatchResultHandler.addHandler(new MethodLabelHandler)
+        dispatchResultHandler.addHandler(new DelegationHandler(delegating.getQuality))
+      case _ =>
+        dispatchResultHandler.addHandler(new ServletResultHandler)
+    }
+    if (configuration.isEnableApiCoverage) {
+      dispatchResultHandler.addHandler(new InstrumentedHandler)
+      dispatchResultHandler.addHandler(new ApiCoverageHandler)
+    }
+    Option(configuration.getDotOutput) match {
+      case Some(dotName) =>
+        if (StringUtils.isNotBlank(dotName)) {
+          val dotPath: String = StringUriUtilities.formatUri(getPath(dotName, configurationRoot))
+          val out: File = new File(dotPath)
+          try {
+            if (out.exists && out.canWrite || !out.exists && out.createNewFile) {
+              dispatchResultHandler.addHandler(new SaveDotHandler(out, true, true))
+            } else {
+              logger.warn("Cannot write to DOT file: " + dotPath)
+            }
+          } catch {
+            case ex: IOException =>
+              logger.warn("Cannot write to DOT file: " + dotPath, ex)
+          }
+        }
+      case _ =>
+    }
+    dispatchResultHandler
+  }
+
+  private def getPath(path: String, configRoot: String): String = {
+    val file: File = new File(path)
+    if (file.isAbsolute) {
+      file.getAbsolutePath
+    } else {
+      new File(configRoot, path).getAbsolutePath
+    }
+  }
+
+  def readResource(resourceStream: InputStream): Option[String] = {
+    Try(Some(Source.fromInputStream(resourceStream).getLines().mkString("\n"))).getOrElse(None)
+  }
+
+  private def rbacToWadl(rbac: Option[String]): Option[String] = {
+
+    case class Resource(path: String, methods: Set[String], roles: Set[String])
+
+    def parseLine(line: String): Option[Resource] = {
+      val values = line.split("\\s+")
+      if (values.length == 3) {
+        Some(new Resource(
+          values(0),
+          Try(values(1).split(',').toSet[String].map(_.trim)).getOrElse(Set.empty),
+          Try(values(2).split(',').toSet[String].map(_.trim)).getOrElse(Set.empty)
+        ))
+      } else {
+        logger.warn(s"Malformed RBAC Resource: $line")
+        None
+      }
+    }
+
+    val parsed = rbac match {
+      case Some(lines) =>
+        Some(lines.replaceAll("[\r?\n?]", "\n").split('\n').toList.flatMap(parseLine))
+      case _ => None
+    }
+
+    parsed match {
+      case Some(values) =>
+        val header = s"""<application xmlns:rax="http://docs.rackspace.com/api"
+                        |         xmlns:xsd="http://www.w3.org/2001/XMLSchema"
+                        |         xmlns="http://wadl.dev.java.net/2009/02"
+                        |    >
+                        |  <resources base="http://localhost">""".stripMargin
+        val resources = values.map { value =>
```

Comment:
You should probably be using scala's built in XML support instead of putting string together the way you are. If anything it will always gurantee that you produce well formed XML -- it'll throw a compile error if you don't.  May may things a bit easier to read too.